pkg/asset/agent/installconfig.go

```diff
@@ -340,22 +341,45 @@ func (a *OptionalInstallConfig) GetBaremetalHosts() []*baremetal.Host {
 func (a *OptionalInstallConfig) validateBMCConfig(installConfig *types.InstallConfig) field.ErrorList {
 	var allErrs field.ErrorList
 
-	bmcConfigured := false
-	for _, host := range installConfig.Platform.BareMetal.Hosts {
-		if host.BMC.Address == "" {
-			continue
+	if a.isBmcConfigured(installConfig) {
+		switch installConfig.Platform.Name() {
+		case baremetal.Name:
+			{
+				fieldPath := field.NewPath("Platform", "BareMetal")
+				allErrs = append(allErrs, baremetalvalidation.ValidateProvisioningNetworking(installConfig.Platform.BareMetal, installConfig.Networking, fieldPath)...)
+			}
+		case none.Name:
+			{
+				fieldPath := field.NewPath("Platform", "None")
+				allErrs = append(allErrs, nonevalidation.ValidateFencingCredentials(installConfig.Platform.None.FencingCredentials, fieldPath)...)
+			}
 		}
-		bmcConfigured = true
-	}
-
-	if bmcConfigured {
-		fieldPath := field.NewPath("Platform", "BareMetal")
-		allErrs = append(allErrs, baremetalvalidation.ValidateProvisioningNetworking(installConfig.Platform.BareMetal, installConfig.Networking, fieldPath)...)
 	}
 
 	return allErrs
 }
 
+func (a *OptionalInstallConfig) isBmcConfigured(installConfig *types.InstallConfig) bool {
```

Comment:
This function is surprising to me since when we last discussed, our implementation was to make `Baremetal.Hosts.BMC` and `fencingCredentials` mutually exclusive. Checking for either seems unintuitive to achieve that goal.

Reply:
Hmm this is basically extension of the original variable `bmcConfigured`.
The way I understand it there isn't a contradiction here, this method merely checks if the bmc is configured either for `baremetal` or  `none` .
In case it does the proper validation applies.
Since this PR is only relevant for `none` platform, BMC can't exist in Hosts for that platform since this API isn't part of `none`.

Reply:
There is no contradiction, but it seems to me like these are seperate, unrelated functions that have been overloaded into a single function.

If we need a function to detect this block I would like to see seperate `areFencingCredentialsConfigured` and `isBmcConfigured`